images/ubuntu2004/README.md

```diff
@@ -0,0 +1,64 @@
+# Ubuntu 20.04 Server
+
+We use packer with cloud-init to create an Ubuntu 20.04 vhdx, which requires [user-data and meta-data files](https://ubuntu.com/server/docs/install/autoinstall-quickstart).
+
+We must also configure the VM using post-install scripts to be suitable to run on [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-ubuntu).
+
+### Create the Image
+
+Follow the steps in the top level [README](../../README.md) to configure and deploy the builder VM. Inside the builder VM, open PowerShell as Administrator and run the following commands to build an image from the `ubuntu2004` sample in this repo.
+
+
+``` powershell
+# Clone the repo
+git clone https://github.com/Azure-Samples/nested-virtualization-image-builder --config core.autocrlf=input
+
+# Build a sample Ubuntu 20.04 image
+cd .\nested-virtualization-image-builder\images\ubuntu2004\
+packer build .\ubuntu2004.pkr.hcl
+
+# Convert outputted vhdx to vhd
+Convert-VHD -Path '.\output-ubuntu2004\Virtual Hard Disks\packer-ubuntu2004.vhdx'  -DestinationPath '.\output-ubuntu2004\Virtual Hard Disks\packer-ubuntu2004.vhd' -VHDType Fixed
```

Comment:
We shouldn't create VHDX and then convert

Set `use_fixed_vhd_format = true` in the packer config to create the correct format directly

Reply:
Also add `skip_compaction = true`

Reply:
Thanks, updated the packer config and removed this line